vendoring/Rust.md

```diff
@@ -0,0 +1,114 @@
+
+```
+%:
+    dh $@ --buildsystem=cargo
+```
+
+Sadly, dh cannot have multiple buildsystems at the same time, so for hybrid codebases you'll need to trigger the build phases manually. Moreover, dh-cargo does *not* support building workspaces at the moment, at it was designed to work on source packages pulled directly from crates.io (which only ships individual crates), so there's a likely chance it'd choke on it.
+
+### Jammy support
+Support for vendored dependencies hasn't been SRUed to Jammy yet, this is tracked in [LP: #2028153](https://bugs.launchpad.net/ubuntu/+source/dh-cargo/+bug/2028153).
+
+## Manual steps
+### `prepare-debian`
+The wrapper expects a few things to be set up before the build:
+
+```make
+override_dh_auto_configure:
+    dh_auto_configure
+    dh_auto_configure --buildsystem=cargo
+```
+
+If you're targeting Jammy, you would want to run something like this instead:
+
+```make
+DEB_HOST_GNU_TYPE=$(DEB_HOST_GNU_TYPE) DEB_HOST_RUST_TYPE=$(DEB_HOST_RUST_TYPE) \
+CARGO_HOME=$(CURDIR)/debian/cargo_home DEB_CARGO_CRATE=adsys_$(shell dpkg-parsechangelog --show-field Version) \
+$(CARGO) prepare-debian $(CARGO_VENDOR_DIR)
+```
+
+### Tests
+Due to the oddities of the Debian Rust ecosystem, by default dh-cargo does not run the tests but rather only does a build test. One needs an override to force it to run the tests.
+
+```make
+override_dh_auto_test:
+    dh_auto_test --buildsystem=cargo -- test --all
+```
+
+### Build
+dh-cargo does *not* build the code in the build stage, it's all folded into the `install` stage.
+
+### Install
+This should work out of the box for single-crate projects:
+
+However, for more complex projects, you'll need this somewhat gnarly invocation along those lines:
+
+```make
+    # Here we utilise a logical flaw in the cargo wrapper to our advantage:
+    # when specifying DEB_CARGO_CRATE_IN_REGISTRY=1, the wrapper will
+    # avoid adding the `--path` option, so that we can specify it ourselves
+    DEB_HOST_GNU_TYPE=$(DEB_HOST_GNU_TYPE) \
+    DEB_HOST_RUST_TYPE=$(DEB_HOST_RUST_TYPE) \
+    CARGO_HOME=$(CURDIR)/debian/cargo_home \
+    DEB_CARGO_CRATE=adsys_mount_0.1.0 \
+    DEB_CARGO_CRATE_IN_REGISTRY=1 \
+    DESTDIR=$(CURDIR)/debian/tmp \
+        $(CARGO) install --path $(ADSYS_MOUNT)
+    # Currently doesn't work because of https://github.com/rust-lang/cargo/issues/4101
+    # combined with a lack of flexibility in the cargo wrapper. That means we
+    # have to do it manually (with the build split out in dh_auto_build for good
+    # measure, even though dh-cargo does it all in the install step)
+    # dh_auto_install --buildsystem=cargo -- --path $(ADSYS_MOUNT)
+```
+
+### Rust vendored sources tracking
+The tracking of embedded Rust vendored dependencies is done via a field in the source package, fittingly named XS-Vendored-Sources-Rust, composed of a comma-separated list of versioned crate names of the format `CRATE@VERSION`.
```

Comment:
Backticks around XS-Vendored-Sources-Rust would be nice, here, `XS-Vendored-Sources-Rust` stands out a bit better.